### ISSUE TITLE
`ruff check --diff` passes when `ruff check` finds an error

### ISSUE THREAD

**Issue by stinodego** [astral-sh/ruff#8584]
Example file:

```python
# repro.py
def hello(x: int):
    """
    Sup.

    Parameters
    ----------
    y
        Hi

    """
```
Running `ruff check repro.py` gives the following output:
```
repro.py:1:5: D417 Missing argument description in the docstring for `hello`: `x`
Found 1 error.
```

Running `ruff check --diff repro.py` **gives no output and passes**.

#### Settings

```toml
# pyproject.toml
[tool.ruff]
select = [
  "D", # flake8-docstrings
]
ignore = [
  "D1",
  "D2",
]
```

Ruff version: `0.1.5`

**Comment by zanieb** [comment]
Hi! This is the expected behavior:

```
      --diff
          Avoid writing any fixed files back; instead, output a diff for each changed file to stdout. Implies `--fix-only`
      --fix-only
          Apply fixes to resolve lint violations, but don't report on leftover violations. 
```

I think what you're looking for will be best resolved by #7352
